src/frontend/next/src/components/SearchResults.tsx

```diff
@@ -0,0 +1,106 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { useSWRInfinite } from 'swr';
+import { Container, Box } from '@material-ui/core';
+
+import useSiteMetadata from '../hooks/use-site-metadata';
+import Timeline from './Posts/Timeline';
+import Spinner from './Spinner';
+
+const useStyles = makeStyles(() => ({
+  spinner: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  searchResults: {
+    padding: 0,
+    width: '100%',
+    justifyContent: 'center',
+  },
+  errorBackground: {
+    position: 'relative',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    borderRadius: '10px',
+    background: '#353F61',
+    boxShadow: '0 15px 30px rgba(0,0,0,.5)',
+    lineHeight: '1rem',
+  },
+  errorTitle: {
+    textAlign: 'center',
+    fontSize: '5rem',
+    color: '#fff',
+  },
+  errorMessage: {
+    lineHeight: '2rem',
+    fontSize: '2rem',
+    textAlign: 'center',
+    margin: '2rem',
+    color: '#96C1E7',
+  },
+}));
+
+type Filter = 'post' | 'author';
+
+type SearchResultProps = {
+  text: string;
+  filter: Filter;
```

Comment:
Get rid of 43 and do this inline, since we don't do anything else with `Filter`:

`filter: 'post' | 'author';`

Reply:
One thing to consider: we might want to define `Filter` like you did on 43 in one file and export it, so we can use it again in another.  I'm not sure who the owner of the type should be, maybe the root search page?  Regardless, that can happen in an other PR.  cc @c3ho, who I think will have to deal with this.